distribution/xfer/download.go

```diff
@@ -70,7 +86,7 @@ type DownloadDescriptor interface {
 	// before).
 	DiffID() (layer.DiffID, error)
 	// Download is called to perform the download.
-	Download(ctx context.Context, progressOutput progress.Output) (io.ReadCloser, int64, error)
+	Download(ctx context.Context, progressOutput progress.Output, cacheDir string) (io.ReadCloser, int64, error)
```

Comment:
@vikstrous The issue here is that it splays the cache dir over the download abstraction. The descriptor itself actually has enough information to calculate the cache dir (as far as I can tell) name if it had a reference to the manager.